### PR TITLE
distrobox: new, 1.4.2.1

### PR DESCRIPTION
**Pull request by YisuiDenghua**
Topic Description
-----------------

Wrapper around podman or docker to create and start containers

Package(s) Affected
-------------------

distrobox

Security Update?
----------------

NO

```
distrobox
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by Artoria2e5** [comment]
Distrobox looks good, not going to approve because I've not been packaging for too long

cwm: is bison a runtime dep or a build dep?

**Comment by OriginCode** [comment]
PR title should be in the format `<package-name>: new, <version>`.

主題 PR 名應使用 `<包名>: new, <版本>` 的格式。

**Review by eatradish — CHANGES_REQUESTED** [review]
Please clean up your commit timeline

`cwm: new, 7.1` is not relevant for this PR

**Comment by YisuiDenghua** [comment]
> Please clean up your commit timeline
> 
> `cwm: new, 7.1` is not relevant for this PR

Thanks, just deleted `cwm: new 7.1`

**Comment by YisuiDenghua** [comment]
> PR title should be in the format `<package-name>: new, <version>`.
> 
> 主題 PR 名應使用 `<包名>: new, <版本>` 的格式。

Fixed, thanks.

**Review by eatradish — DISMISSED** [review]

**Comment by YisuiDenghua** [comment]
> Distrobox looks good, not going to approve because I've not been packaging for too long
> 
> cwm: is bison a runtime dep or a build dep?

Sorry for using Chinese. cwmrc 的 parser 是 bison 生成的

**Comment by Artoria2e5** [comment]
> cwmrc 的 parser 是 bison 生成的

我看了 Makefile, 应该只是编译时依赖，构建的时候 parse.y 变成 parse.c 变成 parse.o，运行的时候用不上

PKGDEP -> BUILDDEP

File: app-emulation/distrobox/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,3 @@
+abinfo "Installing distrobox..."
+bash \
+    -C "$SRCDIR"/install --prefix "$PKGDIR"/usr
```
Comment (cthbleachbit) [inline app-emulation/distrobox/autobuild/build]:
```suggestion
bash "$SRCDIR"/install --prefix "$PKGDIR"/usr
```

需要修正：`-C` 是用于编写补全支持的，不是这样让你调用脚本的。bash 后面可以直接跟脚本文件名

File: app-emulation/distrobox/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,6 @@
+PKGNAME=distrobox
+PKGSEC=utils
+PKGDEP="docker"
+PKGDES="Wrapper around podman or docker to create and start containers"
+ABTYPE=self
```
Comment (cthbleachbit) [inline app-emulation/distrobox/autobuild/defines]: 无需修正：`ABTYPE=self` 在已经手写了 build 的时候可以不写

**Review by cthbleachbit — CHANGES_REQUESTED** [review]

File: app-emulation/distrobox/spec [hunk 3]
```diff
@@ -0,0 +1,4 @@
+VER=1.4.2.1
+SRCS="tbl::https://github.com/89luca89/distrobox/archive/refs/tags/$VER.tar.gz"
```
Comment (CamberLoid) [inline app-emulation/distrobox/spec]: Use `git::https...` source instead

File: app-emulation/distrobox/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=distrobox
+PKGSEC=utils
+PKGDEP="docker"
```
Comment (CamberLoid) [inline app-emulation/distrobox/autobuild/defines]: My opinion: add `bash` as PKGDEP, since this is shell script
```suggestion
PKGDEP="docker bash"
```

**Review by CamberLoid — CHANGES_REQUESTED** [review]

File: app-emulation/distrobox/autobuild/defines (hunk 4 above)
Comment (Artoria2e5) [reply app-emulation/distrobox/autobuild/defines]: 记得她之前问，群里有人说 ab 都需要 bash 不可能没有(
https://t.me/c/1096355081/710708

**Review by cthbleachbit — APPROVED** [review]
Tested to work with podman privileged containers. Rootless podman seems to be broken in itself, so skipping testing with rootless containers...

**Review by CamberLoid — APPROVED** [review]

**Pull request merged**